run_cellregmap.wdl

```diff
@@ -1,209 +1,97 @@
+version development # tells WDL to use most recent version
 
-task GetScatter {
+import "tasks/CellRegMap.wdl" as C
+import "tasks/utils.wdl" as u
+import "tasks/post_processing.wdl" as pp
 
-    input {
-        File featureVariantFile
-        String outputName
-    }
-
-    command {
-        # bash environment
-        python get_scatter.py ${featureVariantFile} --outputName "${outputName}"
-    }
-
-    runtime {
-        memory: ""
-    }
-
-    output {
-        File thisIsMyOutputFile = "this is the output name.txt"
-    }
-}
-
-task RunInteraction {
-    input {
-        Int chrom
-        Float geneName
-        File sampleMappingFile
-        File genotypeFile
-        File phenotypeFile
-        File contextFile
-        File kinshipFile
-        File featureVariantFile
-        Int nContexts = 10
-    }
-
-
-    command {
-        conda activate my_conda_env
-        python run_interaction.py chrom geneName sampleMappingFile genotypeFile phenotypeFile contextFile kinshipFile featureVariantFile nContexts --outputFile ${geneName + ".tsv"}
-    }
-
-    output {
-        File geneOutput = geneName + ".tsv"
-    }
-
-    runtime {
-        # static
-        memory: "400Gb"
-        # # calculated
-        # memory: size(inputFile) + size(interval)
-        # # passed in
-        # memory: memory # from input
-    }
-}
-
-task AggregateInteractionResults {
-    input {
-        Array[File] listOfFiles
-        Float FDR_threshold
-    }
-
-    command {
-        python summarise.py --geneFiles {join(" ", listOfFiles)}
-    }
-
-    output {
-        File all_results
-        File significant_results_only
-    }
-}
-
-
-task EstimateBetas {
-    input {}
-
-    command {}
-
-    output {}
-
-    runtime {
-
-    }
-}
-
-task AggregateBetasResults {
-    input {
-        Array[File] listOfFiles
-    }
-
-    command {
-        python summarise_betas.py --geneFiles {join(" ", listOfFiles)}
-    }
-}
-
-task GetGeneChrPairs {
-    input {
-        File featureVariantsFile
-    }
 
-    command {
-        # does a thing
-        # write a TSV
-        echo 'chr1  BRCA1' > output.tsv
-    }
-
-    output {
-        Array[Array[String, String]] outputPairs = read_tsv("output.tsv")
-    }
-
-}
-
-
-# { WorkflowName.inputName: "value" }
-# { RunCellRegMap.contextFile: "" }
 workflow RunCellRegMap {
     input {
-        File [genotypeFile]
-        File [phenotypeFile]
+        Map[String, File] genotypeFiles # one file per chromosome
+        Map[String, File] genotypeFilesBims
+        Map[String, File] genotypeFilesFams
+        Map[String, File] phenotypeFiles
+        Map[String, File] mafFiles
         File contextFile
-        File covariateFile
         File kinshipFile
         File sampleMappingFile
         File featureVariantFile
-        Array[File] outputFiles
-
-
-        # how to determine which genes to run
-        Array[String] genes
-
-
-        Array[File] intervals
+        Int nContexts=10
+        Float FDR_threshold=1
```

Comment:
Probably drop this to a sensible default

Reply:
are you talking about FDR threshold? i.e. change back from 1 to e.g. 0.05? or am i misunderstanding